### PR TITLE
Proof diffs: don't print the current proof to make_goal_map

### LINKED ISSUE #19612: Tactic used with muligoal selector is fast but goal display update is slow (coqide with proof diffs on)
Original example: compile https://gitlab.inria.fr/fresco/capla/compiler (fork of compcert) and `ProofHelpers` and `abs` from 
[wk.zip](https://github.com/user-attachments/files/17164574/wk.zip), then open `abs_proof.v` from the previous attachment in coqide and enable proof diffs (View > Show Diffs (only added)).
Near the end of the file there is 
~~~coq
  1,2: do 2 f_equal.
  1: do 2 f_equal. 2: do 2 f_equal.
~~~
If you run the `1, 2:` tactic it completes approximately instantly but updating the goal display takes a few seconds.
If you remove it and instead run the 2 single-goal tactics they complete instantly and goal display and also instantly displayed.

Artificial example:
~~~coq
Goal forall a b, S a = S b /\ S b = S a.

  (* use some number high enough for Show Proof to take a noticeable amount of time on your machine *)
  do 4000 (refine (let x := I in _);clear x).
  intros;split.

  all: f_equal. (* completes immediately, but updating goal display takes O(Show Proof) time *)
~~~

### PR DESCRIPTION
**Pull request by SkySkimmer**
Fix #19612

Also makes proof diffs work on #14425 instead of just ignoring the error.

### CODE REVIEW

File: printing/proof_diffs.ml [hunk 1]
```diff
@@ -626,40 +404,30 @@ let map_goal g (osigma, map) = match GoalMap.find_opt g map with
    to create the map from new goal ids to old goal ids.
```
Comment (SkySkimmer) [inline printing/proof_diffs.ml:404]: note to self comment needs updating

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Review by ppedrot — APPROVED** [review]
I don't understand why the previous code was so contrived, but the new variant seems essentially equivalent.

**Comment by ppedrot** [comment]
@coqbot merge now

**Comment by jfehrle** [comment]
> @ppedrot I don't understand why the previous code was so contrived, but the new variant seems essentially equivalent.

It was contrived because the Rocq internals are undocumented, so I had to create something from things I can reverse engineer.

**Pull request merged**